patches/ipq40xx/0019-ipq40xx-support-oap100e.patch

```diff
@@ -0,0 +1,477 @@
++			function = "mdc";
++			bias-pull-up;
++		};
++	};
++};
++
++&cryptobam {
++	status = "okay";
++};
++
++&blsp1_spi1 {
++	pinctrl-0 = <&spi_0_pins>;
++	pinctrl-names = "default";
++	status = "okay";
++	cs-gpios = <&tlmm 12 GPIO_ACTIVE_HIGH>;
++
++	flash@0 {
++		#address-cells = <1>;
++		#size-cells = <1>;
++		compatible = "jedec,spi-nor";
++		reg = <0>;
++		linux,modalias = "m25p80", "gd25q256";
++		spi-max-frequency = <24000000>;
++
++		partitions {
++			compatible = "fixed-partitions";
++			#address-cells = <1>;
++			#size-cells = <1>;
++
++			partition0@0 {
++				label = "0:SBL1";
++				reg = <0x00000000 0x00040000>;
++				read-only;
++			};
++			partition1@40000 {
++				label = "0:MIBIB";
++				reg = <0x00040000 0x00020000>;
++				read-only;
++			};
++			partition2@60000 {
++				label = "0:QSEE";
++				reg = <0x00060000 0x00060000>;
++				read-only;
++			};
++			partition3@c0000 {
++				label = "0:CDT";
++				reg = <0x000c0000 0x00010000>;
++				read-only;
++			};
++			partition4@d0000 {
++				label = "0:DDRPARAMS";
++				reg = <0x000d0000 0x00010000>;
++				read-only;
++			};
++			partition5@e0000 {
++				label = "0:APPSBLENV";
++				reg = <0x000e0000 0x00010000>;
++				read-only;
++			};
++			partition6@f0000 {
++				label = "0:APPSBL";
++				reg = <0x000f0000 0x00080000>;
++				read-only;
++			};
++			partition7@170000 {
++				label = "0:ART";
++				reg = <0x00170000 0x00010000>;
++				read-only;
++			};
++		};
++	};
++};
++
++&nand {
++	pinctrl-0 = <&nand_pins>;
++	pinctrl-names = "default";
++	status = "okay";
++
++	nand@0 {
++		partitions {
++			compatible = "fixed-partitions";
++			#address-cells = <1>;
++			#size-cells = <1>;
++
++			partition@0 {
++				label = "rootfs";
++				reg = <0x00000000 0x4000000>;
++			};
++		};
++	};
++};
++
++&blsp_dma {
++	status = "okay";
++};
++
++&blsp1_uart1 {
++	pinctrl-0 = <&serial_0_pins>;
++	pinctrl-names = "default";
++	status = "okay";
++};
++
++&qpic_bam {
++	status = "okay";
++};
++
++&wifi0 {
++	status = "okay";
++	qcom,ath10k-calibration-variant = "Edgecore OAP100e";
++};
++
++&wifi1 {
++	status = "okay";
++	qcom,ath10k-calibration-variant = "Edgecore OAP100e";
++};
++
++&usb3_ss_phy {
++	status = "okay";
++};
++
++&usb3_hs_phy {
++	status = "okay";
++};
++
++&usb2_hs_phy {
++	status = "okay";
++};
+diff --git a/target/linux/ipq40xx/image/generic.mk b/target/linux/ipq40xx/image/generic.mk
+index c156cb3415..b68c647f04 100644
+--- a/target/linux/ipq40xx/image/generic.mk
++++ b/target/linux/ipq40xx/image/generic.mk
+@@ -414,6 +414,20 @@ define Device/edgecore_oap100
+ endef
+ TARGET_DEVICES += edgecore_oap100
+ 
++define Device/edgecore_oap100e
++	$(call Device/FitImage)
++	$(call Device/UbiFit)
++	DEVICE_VENDOR := Edgecore
++	DEVICE_MODEL := OAP100e
++	SOC := qcom-ipq4019
++	BLOCKSIZE := 128k
++	PAGESIZE := 2048
++	IMAGES := nand-sysupgrade.bin
++	DEVICE_DTS_CONFIG := config@ap.dk07.1-c1
++	DEVICE_PACKAGES := ipq-wifi-edgecore_oap100e kmod-usb-acm kmod-usb-net kmod-usb-net-cdc-qmi uqmi
```

Comment:
@tanyasingh0902 i think the BDF being selected here is not part of the PR

Reply:
@blogic OAP100 and OAP100e have the same BDF and other configs except for the GPIO for external antennas for 5G radio OAP100e. Please suggest what we can do here.

Reply:
@tanyasingh0902 make the devbice select the correct BDF in that case. the current patch selects a non-existing BDF I think.

Reply:
@blogic the BDF of OAP100 is the same as OAP100, so will it be acceptable if we change 
`+	DEVICE_PACKAGES := ipq-wifi-edgecore_oap100e kmod-usb-acm kmod-usb-net kmod-usb-net-cdc-qmi uqmi`
to select OAP100 bdf
`+	DEVICE_PACKAGES := ipq-wifi-edgecore_oap100 kmod-usb-acm kmod-usb-net kmod-usb-net-cdc-qmi uqmi`
?

Reply:
@tanyasingh0902 yes, that would be ok